tests/test_backed_sparse.py

```diff
@@ -270,13 +270,34 @@ def test_dataset_append_memory(
 
 
 @pytest.mark.parametrize("sparse_format", [sparse.csr_matrix, sparse.csc_matrix])
+@pytest.mark.parametrize(
+    ("subset_func", "subset_func2"),
+    product(
+        [
+            ad.tests.helpers.array_subset,
+            ad.tests.helpers.slice_subset,
+            ad.tests.helpers.array_int_subset,
+            ad.tests.helpers.array_bool_subset,
+            lambda x: make_randomized_mask(len(x)),
+            lambda x: make_alternating_mask_15(len(x)),
+            lambda x: make_alternating_mask_5(len(x)),
+            lambda x: make_one_group_mask(len(x)),
+            lambda x: make_one_elem_mask(len(x)),
+        ],
```

Comment:
Should `anndata.tests.helpers.subset_func` be changed to have these parameters?

If no, why are there two different parameter lists of subset functions? Can we put them next to each other and give them speaking names explaining why there are two lists?

Reply:
So these functions were meant to hit (or not) the optimized path within `subset_by_major_axis_mask`, but we don't need to go that deep here so I will these (the coverage in `__getitem__` is the same without these).  They are used in the `test_consecutive_bool` specifically for this purpose where we a mocker/spy to check for optimized path.